qemu/tests/rh_qemu_update.py

```diff
@@ -0,0 +1,308 @@
+import requests
+import json
+import re
+
+from avocado.utils import process
+from virttest import error_context
+
+QUERY_TIMEOUT = 360
+INSTALL_TIMEOUT = 360
+OPERATION_TIMEOUT = 7200
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test update of qemu-kvm
+    1) Boot the VM
+    2) Install qemu-kvm metapackage from compose
+    3) Clone component_management tool
+    4) Verify available, host and guest qemu versions
+    5) Update qemu-kvm using component_management
+    6) Verify installed qemu-kvm version
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+    qemu_package = params['qemu_package_install']
+
+    def install_qemu(session):
+        """
+        Install compose version of qemu-kvm pkg by the name provided in cfg
+        """
+        cmd = "{} install -y {}".format(
+            params['pm_tool'], qemu_package
+        )
+        s, o = session.cmd_status_output(cmd, timeout=OPERATION_TIMEOUT)
+        if s != 0:
+            test.error("Installation of '{}' failed with: {}".format(
+                qemu_package, o
+                )
+            )
+        test.log.info("Installation of '{}' succeeded".format(
+            qemu_package
+            )
+        )
+
+    def install_component_management(session):
+        """
+        Clone component management repository for preparation of necessary
+        repositories containing qemu-kvm build or whole virt module
+        """
+        cmd_clone = "git clone --depth=1 {} -b {}".format(
+            params['component_management_repo'],
+            params['component_management_branch']
+        )
+        s_clone, o_clone = session.cmd_status_output(
+            cmd_clone, timeout=OPERATION_TIMEOUT
+            )
+        if s_clone != 0:
+            test.error("Clonning of '{}' failed with: {}".format(
+                params['component_management_repo'], o_clone
+                )
+            )
+        test.log.info("Clonning of '{}' succeeded".format(
+            params['component_management_repo']
+            )
+        )
+
+    def _compare_qemu_versions(nvr_1, nvr_2):
+        """
+        Compare two qemu vesrion based on their NVRs, function returns tuple
+        containing 3 items: bool (nvr_1 is newer than nvr_2) and both NVRs
+        """
+        def nvr_to_numbers_list(nvr):
+            nvr_list = re.split(
+                "[-.]", nvr.replace("qemu-kvm", "").split("module")[0][1:]
```

Comment:
Here `qemu-kvm` is hardcode,  it means `qemu_package` i think.
And, how do you think about `re.search('%s-(\d+)\.(\d+)\.(\d+)-(\d+)' % qemu_package, nvr).groups()` ?

Reply:
About the `qemu-kvm` part, yes, there is no reason to hardcode it as we have the same information within configuration file.  About the second issue, yes, this is more straight forward approach.

Reply:
I partially reverted and partially changed this lines after today's off-line discussion. It is more generic now and it also handles other special cases we were discussing (z-stream).